protocol/xmrmaker/claim.go

```diff
@@ -99,6 +108,53 @@ func (s *swapState) claimFunds() (ethcommon.Hash, error) {
 	return txHash, nil
 }
 
+// discoverRelayersAndClaim discovers available relayers on the network,
+func (s *swapState) discoverRelayersAndClaim() (ethcommon.Hash, error) {
+	relayers, err := s.Backend.DiscoverRelayers()
+	if err != nil {
+		return ethcommon.Hash{}, err
+	}
+
+	forwarderAddress, err := s.Contract().TrustedForwarder(&bind.CallOpts{})
+	if err != nil {
+		return ethcommon.Hash{}, err
+	}
+
+	calldata, err := getClaimTxCalldata(common.DefaultRelayerCommission, &s.contractSwap, s.getSecret())
+	if err != nil {
+		return ethcommon.Hash{}, err
+	}
+
+	req, err := relayer.CreateSubmitTransactionRequest(
+		s.ETHClient().PrivateKey(),
+		s.ETHClient().Raw(),
+		forwarderAddress,
+		s.contractAddr,
+		calldata,
+	)
+	if err != nil {
+		return ethcommon.Hash{}, err
+	}
+
+	for _, relayer := range relayers {
```

Comment:
Do we know how these relayers are sorted?

Reply:
they're not sorted, it's effectively random. the real answer is it's in order of records returned to us when we do the dht lookup.